Donation/DonationAmountView.swift

```diff
@@ -0,0 +1,200 @@
+/*****************************************************************************
+ * DonationViewController.swift
+ * VLC for iOS
+ *****************************************************************************
+ * Copyright (c) 2018 VideoLAN. All rights reserved.
+ * $Id$
+ *
+ * Authors: Carola Nitz <nitz.carola # googlemail.com>
+ *
+ *
+ * Refer to the COPYING file of the official project for license.
+ *****************************************************************************/
+
+import PassKit
+
+class DonationAmountView: UIStackView {
+
+    private let donationAmount = UITextField(frame: .zero)
+    private let donationTitle = UILabel(frame: .zero)
+    private let separator = UIView(frame: .zero)
+    private let separator2 = UIView(frame: .zero)
+    private var paymentButton: PKPaymentButton?
+    private var buttons: [UIButton]?
+    private let keyboard = UIButton(type: .custom)
+    let closebutton = UIButton(type: .custom)
+    private var donationManager: DonationManager?
+    
+    lazy var numberFormatter = { () -> NumberFormatter in
```

Comment:
would 100% recommend http://github.com/devxoul/Then for some sugar here

Reply:
I'm hesitant to include a thirdparty framework for syntactic sugar only tbh

Reply:
fair - it's one file. maybe vendor the file instead/reimplement the function. the extension itself is 5-6 lines long https://github.com/devxoul/Then/blob/master/Sources/Then/Then.swift

Reply:
I share Caro's opinion here and note that we decided not to merge this PR in this form.

Reply:
I respect it! just wanted to make that syntactic sugar more widely known